sonar/modules/permissions.py

```diff
@@ -135,13 +139,34 @@ def admin_permission_factory(admin_view):
     return superuser_access_permission
 
 
-def files_permission_factory(*kwargs):
-    """Files rest permission factory."""
+def files_permission_factory(obj, action):
+    """Files rest permission factory.
+
+    admin: all
+    moderator: write(organisation only)
+               read(organisation + as anonymous for other)
+    anonymous: write(none)
+               read(record masked, restricted: embargo etc, text never)
+    """
+    # Permission is allowed for all actions.
     if current_app.config.get('SONAR_APP_DISABLE_PERMISSION_CHECKS'):
         return allow_access
 
-    # TODO: Add checks for accessing files and buckets
-    return allow_access
+    if request.view_args.get('pid_value'):
+        (pid, rec) = request.view_args.get('pid_value').data
+        if pid.pid_type == 'doc':
+            from .documents.permissions import DocumentFilesPermission
+            return DocumentFilesPermission.create_permission(obj, action)
+        if pid.pid_type == 'coll':
+            from .collections.permissions import FilesPermission as CollectionFilesPermission
+            return CollectionFilesPermission.create_permission(obj, action)
+        if pid.pid_type == 'org':
+            from .organisations.permissions import OrganisationFilesPermission
+            return OrganisationFilesPermission.create_permission(obj, action)
+        if pid.pid_type == 'depo':
+            from .deposits.permissions import DepositFilesPermission
+            return DepositFilesPermission.create_permission(obj, action)
+    return FilesPermission.create_permission(obj, action)
```

Comment:
Maybe a good idea to create an utils function returning the `FilesPermissions` subclass based on pid_type.
The code should be simplifer 
```python
permission_class = utils_get_permissions_class(pid_type)  // should return FilePermission as default value
return permission_class.create_permission(obj, action)
```

Reply:
I have create a config instead.